src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSession.cs

```diff
@@ -148,6 +162,12 @@ public void FilterOldEvents(List<ProfilerEvent> events)
                 {
                     events.RemoveRange(0, idx + 1);
                 }
+                else
+                {
+                    // last seen event was overwritten
+                    // other events may have been lost as well
+                    this.eventsLost = true;
```

Comment:
I think events may have a sequence number that would allow us to accurately detect if events are lost.  This implementation has a corner case that the previous "last event" is out of the ring buffer, but no event are lost yet.

Reply:
@kburtram I've updated the method of detecting lost events, feel free to take a look!